src/frequenz/sdk/actor/power_distributing/_batteries_status.py

```diff
@@ -46,77 +89,98 @@ async def async_new(
 
         Raises:
             RuntimeError: If any battery has no adjacent inverter.
-
-        Returns:
-            New instance of this class.
         """
-        self: BatteriesStatus = BatteriesStatus.__new__(cls)
-
-        tasks = [
-            BatteryStatusTracker.async_new(
-                id, max_data_age_sec, max_blocking_duration_sec
+        # At first no battery is working, we will get notification when they start
+        # working.
+        self._current_status = Status(working=set(), uncertain=set())
+
+        # Channel for sending results of requests to the batteries
+        request_result_channel = Broadcast[RequestResult]("battery_request_status")
+        self._request_result_sender = request_result_channel.new_sender()
+
+        self._batteries: Dict[str, BatteryStatusTracker] = {}
+
+        # Receivers are needed to create Select object
+        receivers: Dict[str, Receiver[BatteryStatus]] = {}
+
+        for battery_id in battery_ids:
+            channel = _BatteryStatusChannel(battery_id)
+            receivers[channel.name] = channel.receiver
+
+            self._batteries[channel.name] = BatteryStatusTracker(
+                battery_id=battery_id,
+                max_data_age_sec=max_data_age_sec,
+                max_blocking_duration_sec=max_blocking_duration_sec,
+                status_sender=channel.sender,
+                request_result_receiver=request_result_channel.new_receiver(
+                    f"battery_{battery_id}_request_status"
+                ),
             )
-            for id in battery_ids
-        ]
 
-        trackers = await asyncio.gather(*tasks)
-        self._batteries = {tracker.battery_id: tracker for tracker in trackers}
+        self._battery_status_channel = MergeNamed[BatteryStatus](
+            **receivers,
+        )
 
-        return self
+        self._task = asyncio.create_task(self._run())
 
-    def get_working_batteries(self, battery_ids: Set[int]) -> Set[int]:
-        """Get subset of battery_ids with working batteries.
+    async def stop(self) -> None:
+        """Stop tracking batteries status."""
+        await asyncio.gather(
+            *[
+                tracker.stop()  # pylint: disable=protected-access
+                for tracker in self._batteries.values()
+            ]
+        )
+        await cancel_and_await(self._task)
 
-        Args:
-            battery_ids: batteries ids
+    async def _run(self) -> None:
+        """Start tracking batteries status."""
+        while True:
+            try:
+                await self._update_status(self._battery_status_channel)
+            except Exception as err:  # pylint: disable=broad-except
+                _logger.error("BatteriesStatus failed with error: %s. Restart.", err)
 
-        Raises:
-            KeyError: If any battery in the given batteries is not in the pool.
+    async def _update_status(self, status_channel: MergeNamed[BatteryStatus]) -> None:
```

Comment:
maybe more readable to decompose to a `Receiver`, like below,  but it is ok to keep it as it is also.
``` suggestion
    async def _update_status(self, status_channel: Receiver[tuple[str, BatteryStatus]]) -> None:
```

Reply:
I would prefer to leave it `MergeNamed`.
It is clear and shorter.

Reply:
btw,  why not just use `self._battery_status_channel` and not pass anything at all?